returnn/tf/layers/basic.py

```diff
@@ -3310,21 +3346,29 @@ def transform_config_dict(cls, d, network, get_layer):
     :param get_layer:
     """
     super(RepeatLayer, cls).transform_config_dict(d, network=network, get_layer=get_layer)
-    d["repetitions"] = get_layer(d["repetitions"])
+    if isinstance(d["repetitions"], str):
+      d["repetitions"] = get_layer(d["repetitions"])
 
   @classmethod
   def get_out_data_from_opts(cls, name, axis, repetitions, sources=(), **kwargs):
     """
     :param str name:
     :param str axis:
-    :param LayerBase repetitions:
+    :param LayerBase|int repetitions:
     :param list[LayerBase] sources:
     :rtype: Data
     """
     data = get_concat_sources_data_template(sources, name="%s_output" % name)
-    data = data.copy_as_batch_major()
-    axis = data.get_axis_from_description(axis, allow_int=False)
-    data = data.copy_move_axis(axis, data.get_batch_axis(0))
+    if data.have_batch_axis():
+      data = data.copy_as_batch_major()
+    elif isinstance(repetitions, LayerBase):
+      assert repetitions.output.have_batch_axis()
```

Comment:
Why this assert? Just not implemented otherwise? Because in general, should also work without, right?

Reply:
If this is just not implemented, either add such a comment:
```suggestion
      assert repetitions.output.have_batch_axis()  # not implemented
```
Or raise `NotImplementedError`.

Reply:
Ups, that should not have been an assert but part of the if-condition.